### ISSUE TITLE
Freezing Model drops Output Accuracy

### ISSUE THREAD

**Issue by anandcu3** [tensorflow/tensorflow#12450]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Windows 10
- **TensorFlow installed from (source or binary)**:
Binary
- **TensorFlow version (use command below)**:
v1.2 and v1.3 (Same Problem repeats on both)
- **Python version**: 
3.5
- **CUDA/cuDNN version**:
Cuda compilation tools, release 8.0, V8.0.44
cuDNN Version 5.1.10
- **GPU model and memory**:
GeForce GTX 1060 - 6GB

I have a image segmentation network designed to classify roads and obstacles.  I want to freeze the model and serve it as a API. So I used the default TensorFlow tool for freezing the model. After freezing, the output given by the network are completely off and inaccurate.

Here is one sample. 

**The Input Image**

![00016](https://user-images.githubusercontent.com/13032916/29518128-d016402e-8695-11e7-95e5-1fdac3e5b0f6.png)

**Output when tested using ckpt files**
![16_actual](https://user-images.githubusercontent.com/13032916/29518247-3623ea56-8696-11e7-94e0-6eabe13ab066.png)

**Output after Freezing the Model**
![16](https://user-images.githubusercontent.com/13032916/29518258-4087803e-8696-11e7-9a56-4c5d45d7fda1.png)

I have tried to freeze using different versions of tensorflow, but that has not helped. Since the network is performing as excepted when tested against checkpoint, the issue, I think is in the Freeze Models Script. The network uses Batch_normalisation. Could this be the reason for this drop because I saw a couple of issues linked to that of similar nature? How can I avoid that?

**Comment by ali01** [comment]
Are you able to provide code, an example image, and steps for us to reproduce this problem?

**Comment by anandcu3** [comment]
@ali01  I have added more details and code [here](https://stackoverflow.com/questions/45796759/freezing-model-drops-output-accuracy)..

**Comment by Jabba** [comment]
## System information

**Have I written custom code (as opposed to using a stock example script provided in TensorFlow):**
Yes
**OS Platform and Distribution (e.g., Linux Ubuntu 16.04):**
Windows 10
**TensorFlow installed from (source or binary):**
Source
**TensorFlow version (use command below):**
v1.3
**Python version:**
3.5
**CUDA/cuDNN version:**
No GPU, we're trying to deploy on CPU

I've encountered the same problem with freezing the model. I'm working for a company where we want to serve the model in C++, so I've frozen the network and also found differences in network output.

I've double checked the output of the frozen network in C++ and Python and they are the same. However the output is significantly different when comparing it to the model loaded with a checkpoint. There is no workaround for us to continue otherwise than trying older versions of TensorFlow to see if the issue is not there.

We also use Batch Normalization in the network.

**Comment by Jabba** [comment]
I've managed to figure out the issue on our side. Some post-processing steps were frozen, which wasn't clear from our codebase.

**Comment by aselle** [comment]
@anandcu3  have you had any luck resolveing your issue?

**Comment by aselle** [comment]
Nevermind, read the stackoverflow.

**Comment by IvyGongoogle** [comment]
@Jabba @anandcu3  how do you solve it？ can you give some advises?
